resources/shaders/fire.frag

```diff
@@ -114,7 +148,27 @@ void main() {
   noise *= effectMask.y;
 
   // Map noise value to color.
-  vec4 fire = getFireColor(noise);
+  vec4 fire = vec4(0.0);
+  
+  if (uRandomColor)
+  {
+    vec3 baseColor0 = offsetHue(vec3(1.0,0.0,0.0), hash11(uSeed));
+    vec3 baseColor1 = offsetHue(baseColor0, 0.1);
+    vec3 baseColor2 = offsetHue(baseColor1, 0.1);
+
+    vec4 c0 = vec4(baseColor0,uGradient1.a);
+    vec4 c1 = vec4(baseColor0,uGradient2.a);
+    vec4 c2 = vec4(baseColor1,uGradient3.a);
+    vec4 c3 = vec4(baseColor1,uGradient4.a);
+    vec4 c4 = vec4(baseColor2,uGradient5.a);
+
+
+    fire = getFireColorV2(noise,c0,c1,c2,c3,c4);
+  }
+  else
+  {
+    fire = getFireColor(noise);
```

Comment:
I guess you could remove `getFireColor`, rename `getFireColor2` to `getFireColor` and simply use this one here:

```suggestion
    fire = getFireColor(noise, uGradient1, uGradient2, uGradient3, uGradient4, uGradient5);
```